platform/broadcom/rules.mk

```diff
@@ -32,7 +32,10 @@ SONIC_ALL += $(SONIC_ONE_IMAGE) $(SONIC_ONE_ABOOT_IMAGE) \
              $(DOCKER_FPM)
 
 # Inject brcm sai into sairedis
-$(LIBSAIREDIS)_DEPENDS += $(BRCM_SAI) $(BRCM_SAI_DEV) $(LIBSAITHRIFT_DEV)
+$(LIBSAIREDIS)_DEPENDS += $(BRCM_SAI) $(BRCM_SAI_DEV)
+ifeq ($(ENABLE_SYNCD_RPC),y)
```

Comment:
sai server also need libsaithrift

Reply:
@lguohan 
saiserver is a different target which has dependency on libsaithrift, so if you build docker-saiserver it will build libthrift and libsaithrift if ENABLE_SYNCD_RPC is off
Double checked by running:
``` make -f slave.mk target/docker-saiserver-mlnx.gz ```